tests/runner/src/main.rs

```diff
@@ -56,37 +75,19 @@ fn main() {
         // Create a TestCase object with option.envronment and test_opt
         let test_case = TestCase::new(option.environment.clone(), test_opt);
 
-        // TODO: Build this with Test Runner
-        match option.test_name.as_str() {
-            "smoke" => flv_test::tests::smoke::run(fluvio_client, test_case).await,
-            "concurrent" => flv_test::tests::concurrent::run(fluvio_client, test_case).await,
-            //"many_producers" => {
-            //    flv_test::tests::many_producers::run(fluvio_client, option.clone()).await
-            //}
-            _ => panic!("Test not found"),
-        };
+        let test = inventory::iter::<FluvioTest>
+            .into_iter()
+            .find(|t| t.name == test_name.as_str())
+            .expect("Test not found");
+
+        // Run the test
+        (test.test_fn)(fluvio_client.clone(), test_case);
```

Comment:
need error handling?

Reply:
We're catching the panic in a very simple manner at the moment at L44-L47. #880 will introduce more error handling so we can implement reporting